ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml

```diff
@@ -36,18 +37,20 @@ StatusDialog {
         NumberAnimation { duration: 1000; easing.type: Easing.OutExpo; alwaysRunToEnd: true}
     }
 
-    header: AccountsModalHeader {
+    header: Item {
+        height: selector.height
         anchors.top: parent.top
         anchors.topMargin: -height - 18
-        control.popup.width: 512
-        model: root.swapAdaptor.nonWatchAccounts
-        getNetworkShortNames: root.swapAdaptor.getNetworkShortNames
-        formatCurrencyAmount: root.swapAdaptor.formatCurrencyAmount
-        /* TODO: once the Account Header is reworked we simply should be
-        able to use an index and not this logic of selectedAccount being set */
-        selectedAccount: root.swapAdaptor.getSelectedAccount(root.swapInputParamsForm.selectedAccountIndex)
-        onSelectedIndexChanged: {
-            root.swapInputParamsForm.selectedAccountIndex = selectedIndex
+        AccountSelectorHeader {
```

Comment:
Don't need the special model with `colorizedChainPrefixes` as in `ReceiveModal` above?

Reply:
Ah I see, it's in the adaptor .)